### ISSUE TITLE
State of the project?

### ISSUE THREAD

**Issue by promontis** [vpenades/Epsylon.glTF2Toolkit#1]
Hi @vpenades !

First of all, thanks for putting this up as open source! It looks like the best glTF2 implementation so far.
I'm wondering what the state of this project is... 

for the core (https://github.com/vpenades/Epsylon.glTF2Toolkit/tree/master/src/Epsylon.glTF2Toolkit.Core) you wrote _(Seriously outdated)_. Could you highlight what is missing?

Thnx!

Michel

**Comment by vpenades** [comment]
H @Promontis

The project is on hold but not forgotten.

For the moment I don't have time to work on it and I don't know when I'll be able to continue working on it, but it will happen.

I wanted to write a library that could be useful by itself and not simply be a code dump of the gltf schema.

I also wanted to support some popular extensions...

But in the end it proved to be too much for the limited amount of time I have... So my current roadmap is to remove all the experimental code and make it more lightweight. Hopefully with this change, the API surface will be less daunting.

Then, at a later stage I would re reintroduce some of the cool stuff as a layer on top

May I ask for what you need for?

**Comment by vpenades** [comment]
Yes, that's the plan I had in mind... Actually, the code generation part is essentially finished, the only drawback is that working with gltf directly with the generated code is quite horrible.

But, if it's fine to you, I had in mind to move that code to its own project, probably with a different name to reflect it's just the schema and nothing else

**Comment by promontis** [comment]
@vpenades sounds great to me! I've not really looked into great detail what different parts there are within the repository. Could you give an outline what they are? 

How should be go about in splitting the repository up in smaller pieces?

**Comment by vpenades** [comment]
Hi, I'm already working on a private repository, taking all the neccesary parts required to make a streamlined version of the library. Hopefully, Ill be able to move it to public within this week.

The parts I'm going to grab from the current repository are:

- **Epsylon.glTF2Toolkit.CodeGen** : This is the tool that reads the gltf schema and generates the raw source code structure
- **Epsylon.glTF2Toolkit.Core** : From here I'm going to take the Schema2 namespace and a few helper classes, but the API will be essentially the same as you see in that namespace.
- ** 	Epsylon.glTF2Toolkit.Tests** : Associated unit tests.

All other projects in the current namespace were attempts to use gltf in different case scenarios; they been useful to leverage the main API, and I was hopping to build some sort of High Level API to make gltf access easier to do... but in the end, all these APIs are half finished.

Once the new library is available, the access will be direct to the gltf schema.

**Comment by promontis** [comment]
Ah cool! This seems indeed like the parts that form the base.

Give me a shout if you'll need me to have a look 👍

**Comment by vpenades** [comment]
@promontis you can jump straight to https://github.com/vpenades/glTF2Sharp

**Comment by promontis** [comment]
\o/ awesome @vpenades 
Thanks for taking the time to create the new repository.

Do you already know if there are some cases where it is not working?
Maybe we can create a TODO such as https://github.com/ousttrue/DXGLTF?

**Comment by promontis** [comment]
@vpenades Would you be open to create an organisation for the glTF2Sharp projects? I was thinking that it would be a waste, if the other projects would not be included in the repository. For me at least, the primitives builder are a nice asset!

However, it would be nice if we could split the code into multiple repositories, like concluded earlier. Thus, looking at what other open source projects are doing, I came to the conclusion that maybe an organisation would be nice.

GitHub has the 'GitHub Team for Open Source' plan for organisations, which is free. You'd be the admin and perhaps you can add me as contributor. 

So I would suggest to create:
- a glTF2Sharp organisation
- a glTF2Sharp.Core repository, which is the current glTF2Sharp repository, containing the parser (generator)/DOM with tests.
- add more repositories, like the 'glTF2Sharp.DeviceBuffers' (maybe split it out as well) and 'glTF2Sharp.Primitives', on top of the glTF2Sharp.Core

What do you think?

**Comment by vpenades** [comment]
@promontis Not a bad idea, I'll definitely consider it!

It is true that in this repository there's some nice utilities, like the primitives builder you mentioned, but they were highly experimental, and I will probably explore different ways of building assets.

I am not convinced whether these additional utilities might be located in different repositories or not... I have to leverage pros and cons.

Btw, most probably, I will move this "Epsylon.glTF2Toolkit" to private or remove it althogether, mostly because I don't want to confuse people with two different libraries that look almost identical.

So I suggest we move this converstation to the new repo! 😄
